mindsdb_sql_parser/ast/mindsdb/knowledge_base.py

```diff
@@ -146,4 +149,72 @@ def to_tree(self, *args, level=0, **kwargs):
 
     def get_string(self, *args, **kwargs):
         out_str = f'DROP INDEX ON KNOWLEDGE_BASE {self.name.to_string()}'
-        return out_str
\ No newline at end of file
+        return out_str
+
+
+class EvaluateKnowledgeBase(ASTNode):
+    """
+    Evaluate a knowledge base.
+    """
+    def __init__(
+        self,
+        name: Identifier,
+        test_table: Identifier,
+        llm: dict,
+        save_to: Identifier,
+        generate_data: dict = None,
+        params=None,
+        *args,
+        **kwargs):
+        """
+        Args:
+            name: Identifier -- name of the knowledge base.
+            test_table: Identifier -- name of the table to use for testing.
+            llm: dict -- parameters for the LLM to use for evaluation.
+            save_to: Identifier -- name of the table to save the results to.
+            generate_data: dict -- parameters for generating data, if available.
+            params: dict -- additional parameters for the evaluation.
+        """
+        super().__init__(*args, **kwargs)
+        self.name = name
+        self.test_table = test_table
```

Comment:
I think is better to put everything from 'using' into 'params' and mindsdb executor will controll what is required